### ISSUE TITLE
Accessibility for components Icon/Modal/Notification/Button/CheckBox/Dropdown and Table

### ISSUE THREAD

**Issue by xiangmin4ibm** [carbon-design-system/carbon-components-react#1282]
Accessibility for components Icon/Modal/Notification/Pagination/Button/CheckBox/Dropdown and Table.

## Detailed description
Our project used many Carbon React components.  But when we run the IBM AVT test by @ibma/aat 1.5.1 with policies:  IBM_Accessibility_2018_08. we got many violation(g41/g1148/g1055) issues from below components:
 Icon/Modal/Notification/Pagination/Button/CheckBox/Dropdown/Table
1. For the Icon component,  the title part always report g1148 violation error.
```javascript
return _react2.default.createElement(
    'svg',
    _extends({}, props, { 'aria-label': description }),
    _react2.default.createElement(
      'title',
      null,
      description
    ),
    svgContent
  );
```
2. For the Modal, the div.bx--modal-header > button will report g41 violation.  as it mush has the aria-label parameter. and the div.bx--modal-content > div > label will report g1055 violcation, as it need the role and aria-label parameters.
```javascript
var modalButton = React.createElement(
        'button',
        { className: 'bx--modal-close', type: 'button', onClick: onRequestClose },
        React.createElement(Icon, {
          name: 'close',
          className: 'bx--modal-close__icon',
          description: iconDescription
        })
      );

React.createElement(
          'div',
          { className: 'bx--modal-content' },
          this.props.children
        ),
```

3. For the Notification and Pagination components. the button elements will report g41 violation.  as it must has the aria-label parameter. 
```javascript
return React.createElement(
        'button',
        _extends({}, other, { type: type, className: buttonClasses }),
        React.createElement(Icon, {
          description: iconDescription,
          className: iconClasses,
          'aria-label': ariaLabel,
          name: name
        })
      );
```
```javascript
React.createElement(
            'button',
            {
              className: 'bx--pagination__button bx--pagination__button--backward',
              onClick: this.decrementPage,
              disabled: this.props.disabled || statePage === 1 },
            React.createElement(Icon, {
              className: 'bx--pagination__button-icon',
              name: 'chevron--left',
              description: backwardText
            })
          ),
React.createElement(
            'button',
            {
              className: 'bx--pagination__button bx--pagination__button--forward',
              onClick: this.incrementPage,
              disabled: this.props.disabled || statePage === Math.ceil(totalItems / statePageSize) || isLastPage },
            React.createElement(Icon, {
              className: 'bx--pagination__button-icon',
              name: 'chevron--right',
              description: forwardText
            })
          )
```
4. For the other components, they report g1148 issues as they include the Icon component, that should be fixed after the Icon component been fixed.

**What browser are you working in?**
  Chrome

**What version of the Carbon Design System are you using?**
  "carbon-components": "7.19.1",
  "carbon-components-react": "3.14.0",

**What offering/product do you work on? Any pressing ship or release dates we should be aware of?**
   Our project plan to go production in the middle of this month,  hope that can be fixed soon.

**Comment by joshblack** [comment]
Hi @xiangmin4ibm! 👋 Thanks so much for taking the time to make an issue 😄 

For the violations that are related to `aria-label`s on the top-level node, you can just supply the `aria-label` prop to the component to populate that requirement. What specific violations are you seeing on the icon?

**Comment by xiangmin4ibm** [comment]
That below violation for example:
```javascript
Level: violation
	Message: haac.g1148.colorAA
	XPath: /html[1]/body[1]/div[1]/div[1]/div[2]/div[1]/div[3]/div[1]/div[1]/div[1]/div[1]/div[1]/div[1]/div[1]/section[1]/div[2]/button[1]/svg[1]/title[1]
	Snippet: <title>
	Help: https://aat.w3ibm.mybluemix.net/token/64d108c2-e409-44cf-bb21-f7472da44f68/244e5ee7-7b98-4332-ac18-f3923435dc31/latest/doc/w3/help/en-US/idhi_accessibility_check_g1148.html
```

I used the Button component in my code like this
```javascript
<Button className="bx-download-icon" aria-label="download" icon="download" iconDescription="Download ticket" onClick={this.displayDownloadModal}></Button>
```
and the html generate like this:
<img width="1226" alt="download" src="https://user-images.githubusercontent.com/43059022/45256150-edd63300-b3c4-11e8-9cb2-51b8a9431872.png">

The 1148 issue caused as the Button used the Icon component which has the 1148 issue in for the title.

**Comment by xiangmin4ibm** [comment]
Any update for the issue?

**Comment by joshblack** [comment]
@xiangmin4ibm could you help me understand what the issue is? Unfortunately I don't know what the codes correspond to for the a11y violations. It seems like a color contrast issue given the message: `Message: haac.g1148.colorAA`

**Comment by xiangmin4ibm** [comment]
yeah, that's indeed the color contrast issue, but even we change the color and background with the right color contrast, there still have such issue. so don't know how to fix such issue. and can we just remove the title element?

**Comment by joshblack** [comment]
Could you provide a codesandbox that reproduces the a11y errors @xiangmin4ibm? There should have been a link in the issue template when you were making this issue 👍 

If you could also provide full details for the a11y errors, that would also help out a ton. Thanks!

**Comment by carmacleod** [comment]
Happened to notice this issue sitting here, and I thought you might find this [list of DAP/AAT help files](https://github.ibm.com/IBMa/Tools-Rules-HTML/issues/596#issuecomment-2952455) useful.
For example, here are the ones mentioned in this issue:
- [g41](https://cc-rules-dev.w3ibm.mybluemix.net/latest/doc/w3/help/en-US/idhi_accessibility_check_g41.html)
- [g1148](https://cc-rules-dev.w3ibm.mybluemix.net/latest/doc/w3/help/en-US/idhi_accessibility_check_g1148.html)
- [g1055](https://cc-rules-dev.w3ibm.mybluemix.net/latest/doc/w3/help/en-US/idhi_accessibility_check_g1055.html)

(they also have [this wiki](https://github.ibm.com/IBMa/Tools-Rules-HTML/wiki), but g41 is missing)

Also, I just want to mention that, regarding the color contrast violation on the SVG's `<title>` element, the [DAP/AAT has not yet implemented SVG support](https://github.ibm.com/IBMa/Tools-Rules-HTML/issues/6), and so you can ignore that violation.

**Comment by amortiz** [comment]
@joshblack our project is also getting some of these A11y violations written against us for some of the carbon components, here's one, for the label of a disabled carbon Checkbox component:

```
A11y: Minimum contrast of 3.10 with its background is not sufficient to meet WCAG AA requirements for text of size 14px and weight of 400

===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation: 1.4.3 Contrast (Minimum)
IBM Guidance: https://www.ibm.com/able/guidelines/ci162/contrast.html
Test Tool Type: DAP

===== DEFECT DESCRIPTION =====

Content must provide sufficient contrast between text and its background to enable moderately low vision users to read text without contrast-enhancing technology.

===== EXPECTED RESULT =====
https://aat.w3ibm.mybluemix.net/token/a80a8d09-7f5a-480a-adcc-7e22f77482b0/c19cc7e4-e2e6-4475-89d7-533c2bc34979/archives/2018AugustDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1148.html

```

Screenshot:
![image](https://user-images.githubusercontent.com/10033254/52569963-fc8f9e80-2dd7-11e9-80de-853d5d6a5b04.png)

**Comment by carmacleod** [comment]
@amortiz Disabled fields do not have to pass the color contrast check, so you can ignore the DAP violation.
The DAP actually [fixed that](https://github.ibm.com/IBMa/Tools-Rules-HTML/issues/697) in August 2018, but it looks like it was done just after the August 2018 rule set was finalized. This violation will go away as soon as the next DAP rule set is released.

**Comment by dakahn** [comment]
@xiangmin4ibm could you reproduce those errors in a codesandbox for us to test here on our end?
